dev/dev.ex

```diff
@@ -46,4 +46,101 @@ defmodule Dev do
       end
     end
   end
+
+  @ig_contacts [
+    %{ig: ["@durov"], user_id: "0000017c-0000-0000-0000-ac1100040000"}
+  ]
+
+  import Ecto.Query
+
+  # use this function to get a list like above
+  def ig_contacts do
+    T.Matches.Interaction
+    |> where([c], fragment("data ->> 'type'") == "contact_offer")
+    |> group_by([c], c.from_user_id)
+    |> select([c], %{
+      user_id: c.from_user_id,
+      contacts: fragment("array_agg(data -> 'contacts')")
+    })
+    |> T.Repo.all()
+    |> Enum.map(fn %{contacts: contacts} = info ->
+      %{info | contacts: group_contacts(contacts)}
+    end)
+    |> Enum.map(fn %{user_id: user_id, contacts: contacts} ->
+      %{user_id: user_id, ig: contacts["instagram"]}
+    end)
+    |> Enum.reject(fn %{ig: ig} -> is_nil(ig) end)
+    |> Enum.map(fn %{ig: igs} = i -> %{i | ig: filter_valid_igs(igs)} end)
+    |> Enum.reject(fn %{ig: ig} -> ig == [] end)
+  end
+
+  defp filter_valid_igs(igs) do
+    Enum.filter(igs, fn ig ->
+      Regex.match?(
+        ~r/^[A-Za-z0-9_](?:(?:[A-Za-z0-9_]|(?:\.(?!\.))){0,99}(?:[A-Za-z0-9_]))$/,
+        ig |> String.replace(["@"], "") |> String.trim()
```

Comment:
this needs to be run before filtering otherwise we get invalid username like `@durov  `